File: iteration2/src/controllers/AdminController.java
package iteration2.src.controllers;
import iteration2.src.models.Admin;
import iteration2.src.models.Course;
import java.util.HashMap;
import java.util.List;

public class AdminController {
    private Admin currentAdmin;

    public AdminController() {

    }

    public AdminController(Admin currentAdmin) {
        this.currentAdmin = currentAdmin;
    }

    public List<Course> getCourseList(){
        return currentAdmin.getCourseList();
    }

    public HashMap<Integer, String> getConstraints(){
        return currentAdmin.getConstraints();
    }

    public Course createCourse(Course course){
        return currentAdmin.createCourse(course);
    }

    public void editConstraint(HashMap<Integer, String> editedAttributes){
        currentAdmin.editConstraint(editedAttributes);
    }

    public boolean deleteCourse(Course course){
        return currentAdmin.deleteCourse(course);
    }

    public Course findCourseByCourseCode(String courseCode){
        return currentAdmin.findCourseByCourseCode(courseCode);
    }

}

File: iteration3/src/controllers/AdminController.py
class AdminController:
    def __init__(self, currentAdmin):
        self._currentAdmin = currentAdmin

    def getCourseList(self):
        return self._currentAdmin.fetchCourseList()
    
    def getConstraints(self):
        return self._currentAdmin.fetchConstraints()
    
    def createCourse(self, course):
        return self._currentAdmin.createCourse(course)
    
    def deleteCourse(self, course):
        return self._currentAdmin.deleteCourse(course)
    
    def editConstraint(self, editedAttributes):
        return self._currentAdmin.editConstraints(editedAttributes)
    
    def findCourseByCourseCode(self, courseCode):
        return self._currentAdmin.findCourseByCourseCode(courseCode)
    
    def setNotificationToStudentAndAdvisor(self,notification):
        self._currentAdmin.setNotificationToStudentAndAdvisor(notification)
    
    def logout(self):
        self._currentAdmin.logout()